### DISCUSSION TITLE
Should generated dbt contract enforcement tests be included as dagster asset checks?

### DISCUSSION THREAD

**Discussion by justinbricker-acuitymd** [dagster-io/dagster#20490] [Q&A] [answered]
I have added the:

```yml
contract:
    enforced: true
```

to a dbt model that is materialized with a dagster dbt asset, where I have configured `DbtTranslator` to `enable_asset_checks`.

I am able to see asset checks for other tests defined on dbt models and I can see that the contract enforcement tests are running in the output of the materialized tables, but they do not appear as dagster asset checks. 

Should I expect them to?

Dagster Version: `1.5.14`

**Comment by johannkm** [comment, marked as answer]
Hey Justin, we currently don't ingest contracts as asset checks. They're a little different from tests in that they always run when the model is going to execute, versus tests which can run on their own, not be run, etc. Feel free to file an issue for us to add support!

**Reply by justinbricker-acuitymd** [reply]
Thanks Johann, that is a fair point.  I don't know enough about dbt contracts, but it seems it could be a mistake to conflate them with tests/asset_checks.
